Cocoapods/Networking/Requests/Abstract/CCPRequest.m

```diff
@@ -0,0 +1,30 @@
+//
+//  CCPRequest.m
+//  Cocoapods
+//
+//  Created by Gabriel Massana on 7/2/16.
+//  Copyright © 2016 Gabriel Massana. All rights reserved.
+//
+
+#import "CCPRequest.h"
+
+@implementation CCPRequest
+
++ (instancetype)requestForAPI
+{
+    CCPRequest *request = [self request];
+    
+    return request;
+}
+
+- (void)updateRequestWithEndpoint:(NSString *)endpoint
```

Comment:
What does this method do? is that you will have an existing request and then change it?